### PR TITLE
Update to v7.3.17

### LINKED ISSUE #56: Rebuild JLL for Julia 1.10
This package currently fails to load on Julia 1.10:

```
Failed to precompile SDPA [b9a10b5b-afa4-512f-a053-bb3d8080febc] to "/home/pkgeval/.julia/compiled/v1.10/SDPA/jl_GDpYuF".
julia: /opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/local/include/jlcxx/module.hpp:1251: void jlcxx::Module::add_bits(const string&, JLSuperT*) [with T = SDPA::ConeType; JLSuperT = _jl_value_t; std::__cxx11::string = std::__cxx11::basic_string<char>]: Assertion `(((jl_value_t*)(((jl_taggedvalue_t*)((char*)(super) - sizeof(jl_taggedvalue_t)))->header & ~(uintptr_t)15))==(jl_value_t*)(jl_datatype_type))' failed.

[9] signal (6.-6): Aborted
in expression starting at /home/pkgeval/.julia/packages/SDPA/ROHtf/src/SDPA.jl:13
gsignal at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
abort at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x7f724051740e)
__assert_fail at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_ZN5jlcxx6Module8add_bitsIN4SDPA8ConeTypeE11_jl_value_tEEvRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPT0_ at /home/pkgeval/.julia/artifacts/789898d25da6a16dca4a11dfb4bda8ffab5aabc5/lib/libsdpawrap.so (unknown line)
define_julia_module at /home/pkgeval/.julia/artifacts/789898d25da6a16dca4a11dfb4bda8ffab5aabc5/lib/libsdpawrap.so (unknown line)
register_julia_module at /home/pkgeval/.julia/artifacts/0b9e71d0529d5f5890463d4f9efb7171c279cfd0/lib/libcxxwrap_julia.so (unknown line)
```

Apparently rebuilding the JLL against the latest libcxxwrap should fix this: https://github.com/JuliaInterop/CxxWrap.jl/issues/362#issuecomment-1550896266

### PR DESCRIPTION
**Pull request by odow**
I haven't tested, so lets see if anything broke

Closes #56

### CODE REVIEW

**Comment by odow** [comment]
SDPA_jll rebuilding https://github.com/JuliaPackaging/Yggdrasil/pull/6760

**Comment by odow** [comment]
Now it looks like we've broken Windows

**Comment by blegat** [comment]
The failure on CI should hopefully be fixed by https://github.com/JuliaPackaging/Yggdrasil/pull/8130

File: Project.toml [hunk 1]
```diff
@@ -11,9 +11,9 @@ SDPA_jll = "7fc90fd6-dbef-5a6a-93f8-169f2a2e705b"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
-CxxWrap = "0.12"
+CxxWrap = "0.12, 0.13"
 MathOptInterface = "1"
-SDPA_jll = "=700.300.801"
+SDPA_jll = "=700.300.801, =700.301.800"
```
Comment (odow) [inline Project.toml]:
```suggestion
SDPA_jll = "=700.300.801, =700.301.1700"
```

File: Project.toml [hunk 2]
```diff
@@ -11,9 +11,9 @@ SDPA_jll = "7fc90fd6-dbef-5a6a-93f8-169f2a2e705b"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
-CxxWrap = "0.12"
+CxxWrap = "0.12, 0.13"
```
Comment (odow) [inline Project.toml]:
```suggestion
CxxWrap = "0.12, 0.14"
```

File: Project.toml [hunk 3]
```diff
@@ -11,9 +11,9 @@ SDPA_jll = "7fc90fd6-dbef-5a6a-93f8-169f2a2e705b"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
-CxxWrap = "0.12"
+CxxWrap = "0.12, 0.14"
```
Comment (odow) [inline Project.toml]:
```suggestion
CxxWrap = "0.12"
```

File: Project.toml [hunk 4]
```diff
@@ -8,12 +8,14 @@ CxxWrap = "1f15a43c-97ca-5a2a-ae31-89f07a497df4"
...
 SDPA_jll = "7fc90fd6-dbef-5a6a-93f8-169f2a2e705b"
-Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
-CxxWrap = "0.12"
+CxxWrap = "0.14"
+LinearAlgebra = "<0.0.1, 1.6"
 MathOptInterface = "1"
-SDPA_jll = "=700.300.801"
+SDPA_jll = "=700.301.1700"
```
Comment (odow) [inline Project.toml]:
```suggestion
SDPA_jll = "=700.300.1700"
```

**Comment by blegat** [comment]
Maybe there is an issue with the fact that we updated SDPA but not MUMPS while the new version of SDPA uses the new version of MUMPS.

**Comment by odow** [comment]
I have no idea.

**Comment by blegat** [comment]
The only MUMPS-related change I see from v7.3.8 to v7.3.16 is
https://github.com/blegat/SDPA/commit/d0231acc94c7b2e36958eca870bf6bf5affea801#diff-bfa35ecf42cf12cbd937e1688e5a308213948a879405125f6f885e6ea80ec211R110-R114

**Comment by blegat** [comment]
@barche any chance you have already seen such error when updating to CxxWrap v0.14 and have any idea ?

**Comment by barche** [comment]
I'm sorry for the late response, this is a compat issue: CxxWrap 0.14 is based on libcxxwrap_julia_jll 0.11, but SDPA_jll in Yggdrasil is compiled against libcxxwrap_julia_jll version 0.12.

I should be able to release the next CxxWrap (0.15) soon, but this will also contain an updated libcxxwrap, so possibly you will need to rebuild the SPDA_JLL against that. I just tried this branch locally and it works with `CxxWrap#testjll` and a CxxWrap compat for CxxWrap 0.15.

**Comment by odow** [comment]
Ah! For now, I'll rebuild with libcxxwrap_julia_jll@0.11

I guess the problem was an appropriate compat in SDDP_jll

https://github.com/JuliaBinaryWrappers/SDPA_jll.jl/blob/c3fc3fd01a12a410aefd908bf41e168cc9ef02c6/Project.toml#L8

x-ref https://github.com/JuliaPackaging/Yggdrasil/pull/8206

File: Project.toml [hunk 5]
```diff
@@ -8,12 +8,14 @@ CxxWrap = "1f15a43c-97ca-5a2a-ae31-89f07a497df4"
...
 SDPA_jll = "7fc90fd6-dbef-5a6a-93f8-169f2a2e705b"
-Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
-CxxWrap = "0.12"
+CxxWrap = "0.14"
+LinearAlgebra = "<0.0.1, 1.6"
 MathOptInterface = "1"
-SDPA_jll = "=700.300.801"
+SDPA_jll = "=700.300.1700"
```
Comment (odow) [inline Project.toml]:
```suggestion
SDPA_jll = "=700.300.1701"
```

**Comment by odow** [comment]
Looks like we hit https://github.com/JuliaPackaging/Yggdrasil/issues/7904

**Comment by blegat** [comment]
Maybe we can merge since it's still an improvement over master that doesn't work on Julia v1.10 for any OS

**Comment by odow** [comment]
So it took us a year, but we got there in the end. Thanks all for the help.

**Pull request merged**